### ISSUE TITLE
`:doc Type` not working

### ISSUE THREAD

**Issue by reynir** [idris-lang/Idris-dev#2273]
Idris> :doc Type
    Usage is :doc <functionname>

**Comment by david-christiansen** [comment]
This is fairly straightforward to implement and would be a good first project for someone looking to get started working on Idris.

**Comment by RossMeikleham** [comment]
Hey there,

I've been playing around a bit attempting to solve this issue and was wondering if
I could get some tips on how to further proceed? Below is the progress I've made so far.

Thanks

---

The `cmd_doc` function in the `REPLParser.hs` module is causing the `Usage` error message Since "Type" is a reserved word, and fails on `functionname` parsing. So  I added an extra case which checks specifically for "Type":

```Idris
cmd_doc :: String -> P.IdrisParser (Either String Command)
cmd_doc name = do
    let constant = do
        c <- fmap fst P.constant
        eof
        return $ Right (DocStr (Right c) FullDocs)

    let pType = do 
        P.reserved "Type"
        eof
        return $ Right (DocStr (Left $ P.mkName ("Type", "")) FullDocs)

    let fnName = fnNameArg (\n -> DocStr (Left n) FullDocs) name

    try constant <|> try pType <|> try fnName
```

Checking the REPL after implementing this now shows the following:

```
Idris> : doc Type
No documentation for Type
```
Which is expected since `Type` isn't in the docs. 

In the `AbsSyntax.hs` file there's the following definition for `type1Doc`:

```Idris
type1Doc :: Doc OutputAnnotation
type1Doc = (annotate (AnnType "Type" "The type of types, one level up") $ text "Type 1")
```
This is used in the REPL for checking the type of  `Type`

```
Idris> :type Type
Type : Type 1
```
So from this I'm assuming that the correct output for this issue would be:
```
Idris> :doc Type
Type : Type 1
    The type of types, one level up
```
The problem is I'm unsure how to render this from `type1Doc` in the `process` function in
the `REPL.hs` module

**Comment by david-christiansen** [comment]
This was fixed in #2329.
